static/app/components/hovercard.tsx

```diff
@@ -91,17 +93,64 @@ function Hovercard({
         originPoint={arrowData}
         placement={placement}
         className={className}
+        ref={ref}
       >
         {header ? <Header>{header}</Header> : null}
         {body ? <Body className={bodyClassName}>{body}</Body> : null}
       </StyledHovercard>
     </PositionWrapper>
   );
+}
+
+function Hovercard({
+  body,
+  bodyClassName,
+  children,
+  className,
+  containerClassName,
+  header,
+  offset = 12,
+  displayTimeout = 100,
+  tipBorderColor = 'translucentBorder',
+  tipColor = 'backgroundElevated',
+  ...hoverOverlayProps
+}: HovercardProps): React.ReactElement {
+  const {wrapTrigger, isOpen, ...hoverOverlayState} = useHoverOverlay('hovercard', {
+    offset,
+    displayTimeout,
+    isHoverable: true,
+    className: containerClassName,
+    ...hoverOverlayProps,
+  });
+
+  // Nothing to render if no header or body. Be consistent with wrapping the
+  // children with the trigger in the case that the body / header is set while
+  // the trigger is hovered.
+  if (!body && !header) {
+    return <Fragment>{wrapTrigger(children)}</Fragment>;
+  }
 
   return (
     <Fragment>
       {wrapTrigger(children)}
-      {createPortal(<AnimatePresence>{hovercardContent}</AnimatePresence>, document.body)}
+      {createPortal(
+        <AnimatePresence>
+          {isOpen && (
+            <HovercardContent
+              {...{
+                body,
+                bodyClassName,
+                className,
+                tipBorderColor,
+                tipColor,
+                header,
+                hoverOverlayState,
+              }}
+            />
+          )}
```

Comment:
total nit, but the reason I had this pulled into a `hovercardContent` variable is to just keep the indentation tame here

since before this was one line

```tsx
{createPortal(<AnimatePresence>{hovercardContent}</AnimatePresence>, document.body)}
```